Sd 577 replace deprecated snapshotdistributor

filipdujmusic opened the pull request.

githinho commented on contracts/services/FaucetService.sol:
```diff
@@ -2,7 +2,16 @@
-import "./IFaucetService.sol";
+
+interface IFaucetService {
+    function faucet(address payable[] calldata _wallets) external;
```
Not related but maybe we can add events inside interface.

githinho commented on contracts/registry/NameRegistry.sol:
```diff
@@ -84,12 +81,6 @@ contract NameRegistry is INameRegistry {
         emit MapCampaign(msg.sender, name, instance, block.timestamp);
     }
 
-    function mapSnapshotDistributor(string memory name, address instance) external override whitelistedFactoryOnly {
```
Update contract version.

githinho commented:
## ⚠️ Unrelated to this PR and repo ⚠️ 

All changes have to be applied in integration tests. It's a good time for this task: https://linear.app/ampnet/issue/SD-578/use-git-modules-in-integration-test

filipdujmusic commented:
@githinho I've updated all of the services and extracted events/functions/structs in their respective interfaces.

As for the versioning, right now the flow used is as follows:
- some of the contracts are changed in the branch (say contracts A,B and C)
- we open a new pull request -> master
- pull request gets approved and is merged -> master
- after merge, package.json version is increased from X to X+1.
- all of the contracts changed in the merge to master are also configured to point to version X+1
- these changes are pushed directly to master as a "v++" commit
- npm publish is executed on the repo once the v++ commit goes to master

If you have an idea of an improvement on this flow let me know.
Also note that version update on the contract may be incremented directly from say "1.0.2" to "1.0.8" if the last time contract X was changed was in the version "1.0.2" and then hasn't change all until the version "1.0.8" was introduced.
This way we can track what feature was introduced in which version of the codebase.

tl;dr I will update the changed contract versions once the PR is merged to master

githinho commented:
> @githinho I've updated all of the services and extracted events/functions/structs in their respective interfaces.
> 
> As for the versioning, right now the flow used is as follows:
> 
>     * some of the contracts are changed in the branch (say contracts A,B and C)
> 
>     * we open a new pull request -> master
> 
>     * pull request gets approved and is merged -> master
> 
>     * after merge, package.json version is increased from X to X+1.
> 
>     * all of the contracts changed in the merge to master are also configured to point to version X+1
> 
>     * these changes are pushed directly to master as a "v++" commit
> 
>     * npm publish is executed on the repo once the v++ commit goes to master
> 
> 
> If you have an idea of an improvement on this flow let me know. Also note that version update on the contract may be incremented directly from say "1.0.2" to "1.0.8" if the last time contract X was changed was in the version "1.0.2" and then hasn't change all until the version "1.0.8" was introduced. This way we can track what feature was introduced in which version of the codebase.
> 
> tl;dr I will update the changed contract versions once the PR is merged to master

Ok, we definitely need to define some better procedures because this leaves a lot of room for error like not upgrading the version. You can write your ideas to: https://linear.app/ampnet/issue/SD-579/tokenizer-versioning

domagojlatecki commented on deploy.sh:
```diff
@@ -83,10 +80,11 @@ APX_REGISTRY_PRICE_MANAGER=${APX_REGISTRY_PRICE_MANAGER}
 FEE_MANAGER=${FEE_MANAGER}
 FEE_MANAGER_OWNER=${FEE_MANAGER_OWNER}
 FEE_MANAGER_TREASURY=${FEE_MANAGER_TREASURY}
+PAYOUT_MANAGER=${PAYOUT_MANAGER}
```
We should also deploy MerkleTreePathValidator as it is a constructor dependency for PayoutManager (and we may need it elsewhere in the future).

domagojlatecki commented on util/helpers.ts:
```diff
@@ -38,6 +38,18 @@ export async function deployFeeManager(deployer: Signer, owner: string, treasury
 }
 
+export async function deployPayoutManager(deployer: Signer, confirmations: number = config.confirmationsForDeploy): Promise<Contract> {
+  const MerkleTreePathValidator = await ethers.getContractFactory("MerkleTreePathValidator", deployer);
```
We should deploy this separately so that we can use the same contract in the future if we need it.

filipdujmusic replied:
well noted, I'll separate the process in the QueryService PR
